asteroid/binarize.py

```diff
@@ -0,0 +1,162 @@
+        # Start with an empty binary sequence
+        active = []
+        # Check for fully silent or fully voice sequence
+        active, check = check_silence_or_voice(active, pair)
+        if check:
+            return active
+        # Counter for every set of (value, num_consecutive_occ)
+        i = 0
+        # Do until every every sets as been used i.e until we have same sequence length as input length
+        while i < len(pair):
+            # Counter for active set of (value, num_consecutive_occ) i.e (1,num_consecutive_occ)
+            actived = 0
+            # Counter for inactive set of (value, num_consecutive_occ) i.e (0,num_consecutive_occ)
+            not_actived = 0
+            # num_consecutive_occ <  int(stability * sample_rate) need to resolve instability
+            if pair[i][1] < int(stability * sample_rate):
+                # Resolve instability
+                active, i = resolve_instability(
+                    i, pair, stability, sample_rate, actived, not_actived, active
+                )
+            # num_consecutive_occ >  int(stability * sample_rate) we can already choose
+            else:
+                if pair[i][0]:
+                    active.append(torch.ones(pair[i][1]))
+                else:
+                    active.append(torch.zeros(pair[i][1]))
+                i += 1
+        # Stack sequence to return a batch shaped tensor
+        batch_active.append(torch.hstack(active))
+    batch_active = torch.vstack(batch_active).unsqueeze(1)
+    return batch_active
+
+
+def check_silence_or_voice(active, pair):
+    """Check if sequence is fully silence or fully voice.
+
+    Args:
+        active (List) : List containing the binary sequence
+        pair: (List): list of value and consecutive occurrences
+
+    """
+    check = False
+    if len(pair) == 1:
+        check = True
+        if pair[0][0]:
+            active = torch.ones(pair[0][1])
+        else:
+            active = torch.zeros(pair[0][1])
+    return active, check
+
+
+def resolve_instability(i, pair, stability, sample_rate, actived, not_actived, active):
+    """Resolve stability issue in input list of value and num_consecutive_occ
+
+    Args:
+        i (int): The index of the considered pair of value and num_consecutive_occ.
+        pair (list) : Value and num_consecutive_occ.
+        stability (float): Minimal number of seconds to change from 0 to 1 or 1 to 0.
+        sample_rate (int): The sample rate of the waveform.
+        actived (int) : Number of occurrences of the value 1.
+        not_actived (int): Number of occurrences of the value 0.
+        active (list) : The binary sequence.
+
+    Returns:
+        active (list) : The binary sequence.
+         i (int): The index of the considered pair of value and num_consecutive_occ.
+    """
+
+    # Until we find stability count the number of samples active and inactive
+    while i < len(pair) and pair[i][1] < int(stability * sample_rate):
```

Comment:
Again, I think unpacking the pairs here would help readability, and also below with `actived`. Maybe you could even use a `namedtuple`.